src/Sarif/Core/FileData.cs

```diff
@@ -42,9 +70,37 @@ public static FileData Create(Uri uri, bool computeHashes)
                                 Algorithm = AlgorithmKind.Sha256,
                             },
                         };
+                }
+
             }
+            catch (Exception e) when (e is IOException || e is UnauthorizedAccessException) { }
 
             return fileData;
         }
+
+        private static string EncodeFileContents(string filePath, string mimeType, Encoding inputFileEncoding)
+        {
+            byte[] fileContents;
+
+            if (mimeType != SarifWriters.MimeType.Binary)
+            {
+                fileContents = Encoding.UTF8.GetBytes(File.ReadAllText(filePath, inputFileEncoding));
+            }
+            else
+            {
+                fileContents = File.ReadAllBytes(filePath);
+            }
+
+            return Convert.ToBase64String(fileContents);
+        }
+
+        private static Encoding DetectEncoding(string filePath)
```

Comment:
>DetectEncoding [](start = 32, length = 14)

Nobody calls this.

Reply:
I think you intended to say "if inputEncoding not supplied, then detect encoding"

---
In reply to: [136173600](https://github.com/Microsoft/sarif-sdk/pull/612#discussion_r136173600) [](ancestors = 136173600)